src/plugins/intel_gpu/src/graph/impls/ocl/convolution.cpp

```diff
@@ -31,13 +31,19 @@ struct convolution_impl : typed_primitive_impl_ocl<convolution> {
 
         auto outer_id = _outer.id();
         auto data_type = instance.node.input().get_output_layout().data_type;
+        cldnn::data_types weights_data_type;
+        if (instance.get_network().is_internal()) {
+           weights_data_type = instance.node.weights().get_output_layout().data_type;
+        } else {
+           weights_data_type = instance.weights_memory(0)->get_layout().data_type;
```

Comment:
Is this branch really needed? Won't the following work?
`cldnn::data_types weights_data_type = instance.node.weights().get_output_layout().data_type;`

Reply:
I think this is good too. Just made this branch to preseve original path too.. :)

Reply:
Done. Thanks!